web_external/views/map/MapPanel.js

```diff
@@ -283,9 +286,75 @@ const MapPanel = Panel.extend({
             }
         }, this);
 
+        this.listenTo(events, 'm:request-show-bounds', (datasetAndBounds) => {
+            this.removeDatasetBounds();
+            this.datasetAndBounds = datasetAndBounds;
+            this.drawDatasetBounds();
+        });
+
+        this.listenTo(events, 'm:request-remove-bounds', () => {
+            this.datasetAndBounds = null;
+            this.removeDatasetBounds();
+        });
+
+        this.listenTo(events, 'm:toggle-bounds-label', () => {
+            this.showBoundsDatasetLabel = !this.showBoundsDatasetLabel;
+            this.removeDatasetBounds();
+            this.drawDatasetBounds();
+        });
+
         Panel.prototype.initialize.apply(this);
     },
 
+    drawDatasetBounds() {
+        this.annotationLayer = this.map.createLayer('annotation', {
+            annotations: ['rectangle'],
+            showLabels: this.showBoundsDatasetLabel
+        });
+        var i = 0;
+        var features = this.datasetAndBounds.map(({ dataset, bounds }) => {
+            var color = colorbrewer.Set1[9][i++ % 9];
+            return {
+                type: 'Feature',
+                geometry: {
+                    type: 'Polygon',
+                    coordinates: [[
+                        [bounds.lrx, bounds.lry],
+                        [bounds.ulx, bounds.lry],
+                        [bounds.ulx, bounds.uly],
+                        [bounds.lrx, bounds.uly],
+                        [bounds.lrx, bounds.lry]
+                    ]]
+                },
+                properties: {
+                    annotationType: 'rectangle',
+                    name: dataset.get('name'),
+                    labelFontWeight: 'normal',
+                    labelColor: 'grey',
```

Comment:
I find the `grey` has low contrast on many of the background colors.  Perhaps the label should be darker with a light shadow to help highlight it?